Dockerfile

```diff
@@ -6,20 +6,53 @@ RUN mkdir -p /tmp && chown -R www-data:www-data /tmp && chmod -R 777 /tmp
 RUN mkdir -p /var/run/ && chown -R www-data:www-data /var/run && chmod -R 777 /var/run
 USER www-data
 
-FROM base AS prod
+# We install the dependencies in a separate layer as the frontend image also needs them
+FROM base AS vendor
 USER root
-COPY . /var/www/html
+
+# Copying the full context is not great for caching
+# But composer install executes symfony commands that need the full context
+COPY . /var/www/html/
 
 RUN composer install
+
+FROM node:22 AS frontend
+
+WORKDIR /var/www/html
+
+# Needed for some symfony specific modules
+COPY --from=vendor /var/www/html/vendor /var/www/html/vendor
+
+# Only copy what is needed, improves cacheability
+COPY package.json package-lock.json webpack.config.js /var/www/html/
+COPY assets/ /var/www/html/assets/
+
 RUN npm install
 RUN npm run build
 
+FROM base AS prod
+USER root
+
+COPY . /var/www/html/
+
+COPY --from=vendor /var/www/html/vendor /var/www/html/vendor
+COPY --from=frontend /var/www/html/public/build /var/www/html/public/build
+
 RUN chown -R www-data:www-data /var/www/html
```

Comment:
Can you avoid the chmod by copying with the `--chown=ww-data:ww-data`  flag.

Reply:
Oh I did not know that, will definitely try that !

Reply:
done